lib/danom/just.rb

```diff
@@ -15,7 +15,7 @@ class CannotBeNil < StandardError; end
 
     # @raise [Danom::Just::CannotBeNil] if value is provided as nil
     def initialize(value)
-      raise CannotBeNil if value == nil
+      raise CannotBeNil if !(Monad === value) && value == nil
```

Comment:
Danom::Just#initialize performs a nil-check

[Read more about it here.](https://github.com/troessner/reek/blob/master/docs/Nil-Check.md)